### PR TITLE
test-bot: allow `brew doctor` to fail

### PR DESCRIPTION
**Pull request by xu-cheng**
For example, it can be help to prevent [this test](http://bot.brew.sh/job/Homebrew%20Science%20Pull%20Requests/1043/) emitting false alarm.

### CODE REVIEW

**Comment by mikemcquaid** [comment]
Sorry, I don't agree with this change. We should fix `brew doctor` instead if it's emitting incorrect warnings.

**Comment by xu-cheng** [comment]
Then what's your idea to fix something like this:

```
Warning: Some installed formula are missing dependencies.
You should `brew install` the missing dependencies:

    brew install numpy
```

File: Library/Homebrew/cmd/test-bot.rb [hunk 1]
```diff
@@ -549,15 +554,13 @@ def test(*args)
     end
 
     def check_results
-      status = :passed
-      steps.each do |step|
+      steps.all? do |step|
```
Comment (xu-cheng) [inline Library/Homebrew/cmd/test-bot.rb]: @mikemcquaid I did a little refactoring here. Do you want I submit another PR just for this code refactoring.

Comment (mikemcquaid) [inline Library/Homebrew/cmd/test-bot.rb]: Yeh, please.

**Comment by mikemcquaid** [comment]
> Warning: Some installed formula are missing dependencies.
> You should `brew install` the missing dependencies:
> 
>     brew install numpy

Well, why are the dependencies missing? Has the bot failed to cleanup properly? Is the detection failing?

**Comment by xu-cheng** [comment]
I don't know. But it should be related to cross taps dependency.

File: Library/Homebrew/cmd/test-bot.rb (hunk 1 above)
Comment (xu-cheng) [inline Library/Homebrew/cmd/test-bot.rb]: https://github.com/Homebrew/homebrew/pull/36893
